src/main/kotlin/no/nav/tilleggsstonader/sak/behandling/vent/SettPåVentService.kt

```diff
@@ -0,0 +1,182 @@
+import no.nav.tilleggsstonader.sak.behandling.historikk.domain.StegUtfall
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.feilHvis
+import no.nav.tilleggsstonader.sak.infrastruktur.sikkerhet.SikkerhetContext
+import no.nav.tilleggsstonader.sak.opplysninger.oppgave.OppgaveService
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.util.Optional
+import java.util.UUID
+
+@Service
+class SettPåVentService(
+    private val behandlingService: BehandlingService,
+    private val behandlingshistorikkService: BehandlingshistorikkService,
+    private val oppgaveService: OppgaveService,
+    private val settPåVentRepository: SettPåVentRepository,
+) {
+
+    fun hentStatusSettPåVent(behandlingId: UUID): StatusPåVentDto {
+        val settPåVent = settPåVentRepository.findByBehandlingIdAndAktivIsTrue(behandlingId)
+            ?: error("Finner ikke settPåVent for behandling=$behandlingId")
+        val oppgave = oppgaveService.hentOppgave(settPåVent.oppgaveId)
+
+        return StatusPåVentDto(
+            årsaker = settPåVent.årsaker,
+            kommentar = settPåVent.kommentar,
+            frist = oppgave.fristFerdigstillelse,
+            oppgaveVersjon = oppgave.versjonEllerFeil(),
+        )
+    }
+
+    @Transactional
+    fun settPåVent(behandlingId: UUID, dto: SettPåVentDto): StatusPåVentDto {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        feilHvis(behandling.status.behandlingErLåstForVidereRedigering()) {
+            "Kan ikke sette behandling på vent når status=${behandling.status}"
+        }
+        opprettHistorikkInnslag(behandling, StegUtfall.SATT_PÅ_VENT, mapOf("årsaker" to dto.årsaker))
+        behandlingService.oppdaterStatusPåBehandling(behandlingId, BehandlingStatus.SATT_PÅ_VENT)
+
+        val oppdatertOppgave = settOppgavePåVent(behandlingId, dto)
+        val settPåVent = SettPåVent(
+            behandlingId = behandlingId,
+            oppgaveId = oppdatertOppgave.oppgaveId,
+            årsaker = dto.årsaker,
+            kommentar = dto.kommentar,
+        )
+        settPåVentRepository.insert(settPåVent)
+
+        return StatusPåVentDto(
+            årsaker = dto.årsaker,
+            kommentar = dto.kommentar,
+            frist = dto.frist,
+            oppgaveVersjon = oppdatertOppgave.versjon,
+        )
+    }
+
+    @Transactional
+    fun oppdaterSettPåVent(behandlingId: UUID, dto: OppdaterSettPåVentDto): StatusPåVentDto {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        feilHvis(behandling.status != BehandlingStatus.SATT_PÅ_VENT) {
+            "Status på behandlingen må være ${BehandlingStatus.SATT_PÅ_VENT} for å kunne oppdatere"
+        }
+        val settPåVent = settPåVentRepository.findByBehandlingIdAndAktivIsTrue(behandlingId)
+            ?: error("Finner ikke aktiv sett på vent for behandling=$behandlingId")
```

Comment:
Denne går igjen for de opprett/oppdater/slett. Trekke ut i metode for lesbarhet?

Reply:
[1037f90](https://github.com/navikt/tilleggsstonader-sak/pull/227/commits/1037f90ce3b6c8e0fa44cf680b17481995b5d25b)